File: examples/range_estimate.py
from numpy import *
import stochbb
from matplotlib import pylab

Tmin, Tmax, N = 0, 700, 1000;
t = linspace(Tmin, Tmax, N)

stochbb.Logger.addHandler(stochbb.IOLogHandler())

X = stochbb.gamma(10,10);
Y = stochbb.gamma(10,20);
Z = X+Y+100;
print X.density().rangeEst(0.01)
print Y.density().rangeEst(0.01)
print Z.density().rangeEst(0.01)

dX = empty(N,); X.density().eval(Tmin, Tmax, dX);
dY = empty(N,); Y.density().eval(Tmin, Tmax, dY);
dZ = empty(N,); Z.density().eval(Tmin, Tmax, dZ);

pylab.plot(t, dX)
pylab.plot(t, dY)
pylab.plot(t, dZ)

pylab.show()

File: examples/range_estimate.R
library(stochbb)
Tmin <- 0; Tmax <- 700; N <- 1000;
t = seq(Tmin, Tmax, length.out=N)

X <- gamma(10,10);
Y <- gamma(10,20);
Z = X %+% Y %+% 100;

rX <- X$density$rangeEst(0.01);
rY <- Y$density$rangeEst(0.01);
rZ <- Z$density$rangeEst(0.01);

dX <- array(0, c(N)); X$density$eval(Tmin, Tmax, dX);
dY <- array(0, c(N)); Y$density$eval(Tmin, Tmax, dY);
dZ <- array(0, c(N)); Z$density$eval(Tmin, Tmax, dZ);

plot(t, dX, type="l"); abline(v=rX, lty=1)
lines(t, dY, lty=2); abline(v=rY, lty=2)
lines(t, dZ, lty=3); abline(v=rZ, lty=3)

